File: src/lib/api/notify.ts
import { invoke } from "@tauri-apps/api/core";

export const notifyApi = {
  /** Sends through the native macOS UNUserNotificationCenter path (correct
   * icon + click-to-open). Returns false when unavailable — non-macOS, or a
   * dev binary with no app bundle — so callers can fall back to the
   * notification plugin. */
  native(title: string, body: string, targetPath: string): Promise<boolean> {
    return invoke("notify_native", { title, body, targetPath });
  },
  /** Returns and clears the path of a notification clicked before the
   * frontend was listening (app launched by the click). */
  takePendingClick(): Promise<string | null> {
    return invoke("take_pending_notification_click");
  },
};

File: src-tauri/src/notify.rs
//! Cross-platform wrappers over the macOS-only native notification path.
//! On other platforms (and in dev mode, where there's no app bundle) these
//! report "not handled" and the frontend falls back to
//! tauri-plugin-notification.

#[tauri::command]
pub fn notify_native(title: String, body: String, target_path: String) -> Result<bool, String> {
    #[cfg(target_os = "macos")]
    {
        Ok(crate::notify_mac::send(&title, &body, &target_path))
    }
    #[cfg(not(target_os = "macos"))]
    {
        let _ = (title, body, target_path);
        Ok(false)
    }
}

#[tauri::command]
pub fn take_pending_notification_click() -> Option<String> {
    #[cfg(target_os = "macos")]
    {
        crate::notify_mac::take_pending()
    }
    #[cfg(not(target_os = "macos"))]
    {
        None
    }
}
